packages/relay/src/lib/clients/sdkClient.ts

```diff
@@ -643,60 +643,111 @@ export class SDKClient {
 
     const fileCreateTx = await new FileCreateTransaction()
       .setContents(hexedCallData.substring(0, this.fileAppendChunkSize))
       .setKeys(client.operatorPublicKey ? [client.operatorPublicKey] : []);
-    const fileCreateTxResponse = await fileCreateTx.execute(client);
-    const { fileId } = await fileCreateTxResponse.getReceipt(client);
-
-    const createFileRecord = await fileCreateTxResponse.getRecord(this.clientMain);
-    this.captureMetrics(
-      SDKClient.transactionMode,
-      fileCreateTx.constructor.name,
-      Status.Success,
-      createFileRecord.transactionFee.toTinybars().toNumber(),
-      createFileRecord?.contractFunctionResult?.gasUsed,
-      callerName,
-      interactingEntity,
-    );
 
-    if (fileId && callData.length > this.fileAppendChunkSize) {
-      const fileAppendTx = await new FileAppendTransaction()
-        .setFileId(fileId)
-        .setContents(hexedCallData.substring(this.fileAppendChunkSize, hexedCallData.length))
-        .setChunkSize(this.fileAppendChunkSize)
-        .setMaxChunks(this.maxChunks);
-      await fileAppendTx.execute(client);
+    try {
+      const shouldLimit = this.hbarLimiter.shouldLimit(currentDateNow, SDKClient.transactionMode, callerName);
+      if (shouldLimit) {
+        throw predefined.HBAR_RATE_LIMIT_EXCEEDED;
+      }
+      const fileCreateTxResponse = await fileCreateTx.execute(client);
+      const { fileId } = await fileCreateTxResponse.getReceipt(client);
+      createFileId = fileId;
 
+      const createFileRecord = await fileCreateTxResponse.getRecord(this.clientMain);
       this.captureMetrics(
         SDKClient.transactionMode,
-        fileAppendTx.constructor.name,
+        fileCreateTx.constructor.name,
         Status.Success,
-        await this.calculateFileAppendTxTotalTinybarsCost(fileAppendTx),
-        0,
+        createFileRecord.transactionFee.toTinybars().toNumber(),
+        createFileRecord?.contractFunctionResult?.gasUsed,
         callerName,
         interactingEntity,
       );
-    }
 
-    // Ensure that the calldata file is not empty
-    if (fileId) {
-      const fileSize = await (await new FileInfoQuery().setFileId(fileId).execute(client)).size;
+      if (fileId && callData.length > this.fileAppendChunkSize) {
+        const fileAppendTx = await new FileAppendTransaction()
+          .setFileId(fileId)
+          .setContents(hexedCallData.substring(this.fileAppendChunkSize, hexedCallData.length))
+          .setChunkSize(this.fileAppendChunkSize)
+          .setMaxChunks(this.maxChunks);
+        await fileAppendTx.execute(client);
+
+        this.captureMetrics(
+          SDKClient.transactionMode,
+          fileAppendTx.constructor.name,
+          Status.Success,
+          await this.calculateFileAppendTxTotalTinybarsCost(fileAppendTx),
+          0,
+          callerName,
+          interactingEntity,
+        );
+      }
+
+      // Ensure that the calldata file is not empty
+      if (fileId) {
```

Comment:
Q: Where's the actual logic to add the costs of the FileCreate and FileAppend to the hbar rate limiter?